# Review of expressions and variables in Python

This document contains **15 exercises, from basic to advanced,** to practice Python expressions and variables.

## Basic level:

### Exercise 1
Create a variable called `a`and assign it the result of `5 + 7`. Print `a`.

In [139]:
a = 5 + 7
print(a)

12


### Exercise 2
Create a variable called `b` and assign it the result of `20 - 4`. Print `b`.

In [140]:
b = 20 - 4
print(b)

16


### Exercise 3
Create a variable called `c` and assign it the result of `3 * 8`. Print `c`.

In [141]:
c = 3 * 8
print(c)

24


### Exercise 4
Create a variable called `d` and assign it the result of `40 / 5`. Print `d`.

In [142]:
d = 40 / 5
print(d)

8.0


### Exercise 5
Use integer division to divide `17` by `3`and store the result in a variable `e`. Print `e`.

In [143]:
e = 17 // 3
print(e)

5


## Intermediate level

### Exercise 1
Calculate the remainder when 245 is divided by 13. Store it in `remainder` and print it.

In [1]:
remainder = 245 % 13
print(remainder)

11


### Exercise 2
Convert 3 hours and 45 minutes into total minutes. Store it in `total_minutes` and print it.

In [3]:
hours = 3
minutes = 45
total_minutes = hours * 60 + minutes
print(total_minutes)

225


### Exercise 3
Calculate the average of five test scores: 87, 92, 76, 81 and 95. Store in `average_score`.

In [5]:
import numpy as np

test_scores = [87, 92, 76, 81, 95]
array_test_scores = np.array(test_scores)
average_score = np.mean(array_test_scores)
print(f'The mean of the test scores is: {average_score}')

The mean of the test scores is: 86.2


### Exercise 4
Using compound assignment, increase the value of `points`by 50, then multiply by 2. Print final value.

In [6]:
points = 100
points += 50
points *= 2
print(points)

300


### Exercise 5
Suppose a user runs 7 km each day for 4 days. Calculate total kilometers run in a week.

In [7]:
km_day = 7
days = 4
km_week = days * km_day
print(f'Total kilometers run in a week: {km_week}')

Total kilometers run in a week: 28


## Advanced level

### Exercise 1
Create a NumPy array with the finish times (in seconds) of 5 runners in a race. Convert these times to minutes (as floats) and store them in a new array.

In [5]:
import numpy as np

finish_times_s = np.array([52, 123, 352, 78])
finish_times_m = finish_times_s / 60

print(finish_times_m)

[0.86666667 2.05       5.86666667 1.3       ]


### Exercise 2
Using the compound interest formula, calculate the future value of an investment of €1000 at 5% annual interest over 10 years.

In [8]:
def final_amount(P: float, r: float, t: float) -> float:
    """ 
    Calculate the future value of an investment using compound interest.

    Parameters:
        P (float): Initial investment amount.
        r (float): Annual interest rate as a decimal.
        t (float): Time in years.

    Returns:
        float: Final amount after applying compound interest.
    """
    return P * (1 + r) ** t

P = 1000
r = 0.05
t = 10

print(f"Final amount after {t} years: €{final_amount(P, r, t):.2f}")

Final amount after 10 years: €1628.89


### Exercise 3
Create a NumPy array of temperatures in Fahrenheit and convert all values to Celsius using the formula. Do this without using loops.

In [11]:
temperatures_fahrenheit = np.array([42, 123, 241, 32, 111])
temperatures_celsius = np.round((temperatures_fahrenheit - 32) * 5 / 9, 2)
print(temperatures_celsius)

[  5.56  50.56 116.11   0.    43.89]


### Exercise 4
Given a list of prices, use NumPy to calculate the standard deviation. Then determine whethere the prices have high variability based on the result.

In [13]:
prices = np.array([1.5, 2.7, 10.1, 12.1, 1.6])
print(np.round(np.std(prices), 2))

4.55


The calculated standard deviation is 4.55, which is relatively high compared to the average price. This indicates a significant level of variability in the data. In other words, the prices are not tightly clustered around the mean, instead there are some values like 10.1 and 12.1 than deviate considerably, increasing the overall dispersion.

### Exercise 5
Define three variables representing distances (in km) traveled by three vehicles. Calculate the total distance and the average distance.

In [18]:
first_distance = 58
second_distance = 456
third_distance = 1092
array_distances = np.array([first_distance, second_distance, third_distance])

print(f'Total distance tavelled: {np.sum(array_distances)}')
print(f'Average distance travelled: {np.round(np.mean(array_distances), 2)}')

Total distance tavelled: 1606
Average distance travelled: 535.33
